### PR TITLE
sidekiqctl status

### PR DESCRIPTION
**Pull request by dzunk**
Inspired by [this tweet](https://twitter.com/mperham/status/1053350655652327424). Adds `sidekiqctl status` for a CLI status display.

```
$> bin/sidekiqctl status
Sidekiq 5.2.3
2018-10-22 16:43:39 -0700

---- Overview ----
  Processed: 2349
     Failed: 906
       Busy: 0
   Enqueued: 6158
    Retries: 0
  Scheduled: 0
       Dead: 151

---- Processes (2) ----
Dzunks-MacBook-Pro.local:39560:3928e76d6d0b [tag]
  Started: 2018-10-21 12:35:37 -0700 (28 hours ago)
  Threads: 3 (0 busy)
   Queues: foo

Dzunks-MacBook-Pro.local:82458:35a2c5625869 [tag] [quiet]
  Started: 2018-10-21 19:37:26 -0700 (21 hours ago)
  Threads: 5 (0 busy)
   Queues: default, a_queue, another_queue, so_many_queues, wow_lots_of_queues,
           arent_this_many_queues_a_bad_thing, why_so_queued, four_queue_and_seven_queues,
		   queue_washington, critical

---- Queues (4) ----
NAME                     SIZE   LATENCY
a_queue                     0      0.00
another_queue               3  62096.93
so_many_queues           6133  75971.34
default                     0      0.00
```
You can also specify a single section.
```
$> bin/sidekiqctl status overview
---- Overview ----
  Processed: 2349
     Failed: 906
       Busy: 0
   Enqueued: 6158
    Retries: 0
  Scheduled: 0
       Dead: 151
```
Needs cleanup/refactoring/tests, but since the tests will depend on what the output looks like, I wanted to throw it up for some feedback first.

### CODE REVIEW

**Comment by mperham** [comment]
What do you think about output formats? Yours is more readable but less parsable. I know that people will run this and want to parse the output. `redis-cli info` is another possible format.

**Comment by dzunk** [comment]
I was working on a `--format json` to switch between human readable and machine parseable, but that path came dangerously close to reimplementing `sidekiq/api`. I figured if someone is trying to programmatically interact with the data they should be using the API directly. This version is more in the direction of the `passenger-status` command.

As a human, I have a weak preference for readable over parseable 😉 The `redis-cli` style would be a lot easier to implement though, and if the main use case is parsing for use in other scripts that would make more sense.

**Comment by mperham** [comment]
I think that sounds reasonable, you're right that the API should provide any programmatic need. Having something which can blast out a big blob of JSON for [jq usage](https://stedolan.github.io/jq/manual/v1.5/) in scripts seems like a nice idea but ultimately this is your contribution and awesome already.

File: lib/sidekiq/status.rb [hunk 1]
```diff
@@ -0,0 +1,130 @@
+require 'sidekiq/api'
+
+module Sidekiq
+  class Status
```
Comment (mperham) [inline lib/sidekiq/status.rb]: There is already a gem that uses the name "sidekiq-status" so I'd prefer another name.  Given that this is not an API designed for programmatic use, I think all this code should go back into `bin/sidekiqctl` and namespaced in `Sidekiqctl::Status`.

**Comment by mperham** [comment]
Does it work with a non-default Redis, using REDIS_URL?  I think it should, using Sidekiq::RedisConnection.

REDIS_URL=redis://localhost:6379/1 bin/sidekiqctl status

**Comment by dzunk** [comment]
Yep!
```
$> bin/sidekiqctl status
Sidekiq 5.2.2
2018-10-27 19:30:43 +0000

---- Overview ----
  Processed: 0
     Failed: 0
       Busy: 0
   Enqueued: 0
    Retries: 0
  Scheduled: 0
       Dead: 0

---- Processes (0) ----

---- Queues (0) ----
NAME    SIZE  LATENCY
```
```
$> REDIS_URL=redis://redis-staging.weedmaps.com:6379/0 bin/sidekiqctl status
Sidekiq 5.2.2
2018-10-27 19:30:50 +0000

---- Overview ----
  Processed: 190272138
     Failed: 79103669
       Busy: 0
   Enqueued: 0
    Retries: 0
  Scheduled: 0
       Dead: 9999

---- Processes (21) ----
ip-10-0-117-115:22269:a25d4cd351fd [www]
  Started: 2018-10-24 19:04:25 +0000 (72 hours ago)
  Threads: 5 (0 busy)
** snip **
```
Also expanded error handling to cover bad Redis config and nonexistent sections:
```
$> REDIS_URL=redis://example.com:1234/5 bin/sidekiqctl status
Sidekiq 5.2.2
2018-10-27 13:05:11 -0700

---- Overview ----
Couldn't get status: Error connecting to Redis on example.com:1234 (Redis::TimeoutError)
```
```
$> bin/sidekiqctl status foo
I don't know how to check the status of 'foo'!
Try one of these: all, version, overview, processes, queues
```

**Comment by mperham** [comment]
LGTM, all done?

**Comment by dzunk** [comment]
Yep, any README updates necessary/desired?

Matt
________________________________

**Comment by mperham** [comment]
Add a Changelog entry to give yourself credit.

**Comment by mperham** [comment]
Thank you for the excellent contribution!

**Pull request merged**